### PR TITLE
Switched to org.immutables for the POJOs

### PR DESCRIPTION
**Pull request by quintesse**
Not really important but I like the simplicity of org.immutables :-)

### CODE REVIEW

**Comment by cmoulliard** [comment]
Can you please have a look ? @BarDweller

**Comment by BarDweller** [comment]
I can't help but wonder what these would look like using Java Records instead, they are also immutable, and require minimal declarative syntax, although they would mean a minimum jvm level that supports them.

**Comment by cmoulliard** [comment]
> I can't help

Why do you say that ? @BarDweller

**Comment by BarDweller** [comment]
> > I can't help
> 
> Why do you say that ? @BarDweller

I said 'I can't help but wonder' not 'I can't help' .. very different meanings =) 

And my thinking here is that if we're moving from one way of defining an immutable used as a read-only pojo, to another way of doing the same with a different library, that it would be interesting to see how the new Java built in way of doing that compares. 

This mostly comes off the back of experimenting with records as data objects during advent of code, where they really went up in my estimation of how useful they are. But those usages were less real world, and that's what makes me think when looking at this changeset, about how it might look if it used records. Immediate thought go to the construction pattern.. records don't use a builder style pattern, but I suspect that could be emulated.. but I think there's probably a balance point where the more you try to do that, the more boilerplate you end up adding back to something you were trying to avoid being verbose. 

I think my longterm view here is that having used Records, and read some _amazing_ code that used them too, I suspect that may become the direction for future immutable pojo type usage in Java.. I mean, you don't need lombok anymore if you just want an easily defined compound map key, a Record does it, and does it without a dependency being added to a project. 

Anyways, that's what was running through my head as I read the changes =)

**Comment by quintesse** [comment]
@BarDweller I definitely agree with you on the records, and that was actually what I started out with, but I ran into problems where some of the tooling didn't support it yet, so I decided to keep the changes as minimal as possible.

But if we can get it to work and if we can be sure that having Java 14 as a minimum version is acceptable then I would be all for using records. If not then I think this is a good alternative (until the moment we can safely switch to Java 14+)

**Comment by quintesse** [comment]
PS: Actually Java 16, I thought it was 14 but records were only a preview feature back then.

**Comment by quintesse** [comment]
>  but I ran into problems where some of the tooling didn't support it yet

Just checked again and it was basically because there wasn't an official UBI container image yet for Java 16+ at that time, but there's a Java 17 image now so let me try again and see if we can get records to work :-)

**Comment by quintesse** [comment]
Another issue back then was that the builder library [sundrio](https://github.com/sundrio/sundrio) didn't work with records either, but that was implemented in the meantime as well: https://github.com/sundrio/sundrio/issues/259

So I created a different PR using records: #10 

So I would suggest closing this PR and consider applying that one.
